modin/experimental/core/storage_formats/hdk/query_compiler.py

```diff
@@ -647,7 +649,11 @@ def mul(self, other, **kwargs):
         return self._bin_op(other, "mul", **kwargs)
 
     def mod(self, other, **kwargs):
-        return self._bin_op(other, "mod", **kwargs)
+        int_codes = np.typecodes["AllInteger"]
+        if all(t.char in int_codes for t in self._modin_frame.dtypes):
+            return self._bin_op(other, "mod", **kwargs)
```

Comment:
Don't we have a test for this branch?

Reply:
There are tests in test_dataframe, but they were broken. Fixed now.